### PR TITLE
scripts: west_commands: nrfjprog: Add support for --qspiini

### PR DESCRIPTION
**Pull request by carlescufi**
Allow for users to provide a --qspiini parameter that is passed directly to the nrfjprog executable but only in the --program operation. This is required since e073210ec299516476c2b1f1a6b270fb5f7839ca enabled the -O/--tool-opt for all operations, but --qspiini is only allowed combined with --program.

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

File: scripts/west_commands/runners/nrfjprog.py [hunk 1]
```diff
@@ -81,6 +95,9 @@ def do_exec_op(self, op, force=False):
             if _op.get('verify'):
                 # In the future there might be multiple verify modes
                 cmd.append('--verify')
+            if self.qspi_ini:
```
Comment (nvlsianpu) [inline scripts/west_commands/runners/nrfjprog.py:98]: @carlescufi wouldn't this add --qspiint option to any call to nrfjprog in a row within one `west flash` proceddure?

**Review by nordicjm — CHANGES_REQUESTED** [review]
This should only be applied when it is programming the application core on nRF53 devices, if programming both cores then it fails with:
```
-- west flash: using runner nrfjprog
-- runners.nrfjprog: reset after flashing requested
Using board 1050043461
-- runners.nrfjprog: Flashing file: /tmp/bb/nrf/samples/nrf5340/extxip_smp_svr/_AA/merged_CPUNET.hex
ERROR: The operation attempted is unavailable in your device.
NOTE: For additional output, try running again with logging enabled (--log).
NOTE: Any generated log error messages will be displayed.
FATAL ERROR: command exited with status 17: nrfjprog --program /tmp/bb/nrf/samples/nrf5340/extxip_smp_svr/_AA/merged_CPUNET.hex --sectorerase --verify --qspiini ../Qspi_thingy53.ini -f NRF53 --coprocessor CP_NETWORK --snr 1050043461
```

**Review by nordicjm — APPROVED** [review]
Not an issue, can be put in application CMakeLists.txt to only apply to that though might make sense to only supply this argument on cores that support it so it can be used globally

File: scripts/west_commands/runners/nrfjprog.py (hunk 1 above)
Comment (carlescufi) [reply scripts/west_commands/runners/nrfjprog.py:98]: No, because this is only in the `elif op_type == 'program':` section.

**Pull request merged**